### PR TITLE
Add TargetUserName field to "Multiple Password resets by User" and "Process" field to "Excessive Windows logon failures" 

### PR DESCRIPTION
**Pull request by jonyluke**
I think it would be a good option to add the "TargetUserName" field to the "Multiple Password resets by User" rule, in order to see which account the password was tried to change at a glance.

Also, to add the "Process" field to "Excessive Windows logon failures" rule.

### CODE REVIEW

File: Detections/SecurityEvent/ExcessiveLogonFailures.yaml [hunk 1]
```diff
@@ -75,4 +75,8 @@ entityMappings:
     fieldMappings:
       - identifier: FullName
         columnName: HostCustomEntity
-version: 1.0.0
\ No newline at end of file
+  - entityType: Process
+    fieldMappings:
+      - identifier: CommandLine
+        columnName: Process
+version: 1.0.0
```
Comment (ashwin-patil) [inline Detections/SecurityEvent/ExcessiveLogonFailures.yaml:82]: please change the version to 2.0.0.

**Review by ashwin-patil — CHANGES_REQUESTED** [review]
Need version change, rest looks good. Thank you for submitting PR.

**Review by ashwin-patil — APPROVED** [review]

**Comment by jonyluke** [comment]
Is it necessary any more changes or can the PR be closed? @ashwin-patil

**Review by ashwin-patil — CHANGES_REQUESTED** [review]
I have found some additional bugs with this query.. Adding the updated query including your proposed changes.. Please review and update PR accordingly..

Additional changes Added:
- Renamed `TargetResourceName` with `TargetUserName` in auditlogs
- replaced `makeset` with `make_set`
- in the output created additional field TargetUserList containing list of accounts
- created field TargetUserName selected recent account in order to assign entity

Final Query:
```
  let PerUserThreshold = 5;
  let TotalThreshold = 100;
  let action = dynamic(["change", "changed", "reset"]);
  let pWord = dynamic(["password", "credentials"]);
  let PasswordResetMultiDataSource =
  (union isfuzzy=true
  (//Password reset events
  //4723: An attempt was made to change an account's password
  //4724: An attempt was made to reset an accounts password
  SecurityEvent
  | where EventID in ("4723","4724")
  | project TimeGenerated, Computer, AccountType, Account, Type, TargetUserName),
  (//Azure Active Directory Password reset events
  AuditLogs
  | where OperationName has_any (pWord) and OperationName has_any (action)
  | extend AccountType = tostring(TargetResources[0].type), Account = tostring(TargetResources[0].userPrincipalName), 
  TargetUserName = tolower(tostring(TargetResources[0].displayName))
  | project TimeGenerated, AccountType, Account, Computer = "", Type),
  (//OfficeActive ActiveDirectory Password reset events
  OfficeActivity
  | where OfficeWorkload == "AzureActiveDirectory" 
  | where (ExtendedProperties has_any (pWord) or ModifiedProperties has_any (pWord)) and (ExtendedProperties has_any (action) or ModifiedProperties has_any (action))
  | extend AccountType = UserType, Account = OfficeObjectId 
  | project TimeGenerated, AccountType, Account, Type, Computer = ""),
  (// Unix syslog password reset events
  Syslog
  | where Facility in ("auth","authpriv")
  | where SyslogMessage has_any (pWord) and SyslogMessage has_any (action)
  | extend AccountType = iif(SyslogMessage contains "root", "Root", "Non-Root")
  | where SyslogMessage matches regex ".*password changed for.*"
  | parse SyslogMessage with * "password changed for" Account
  | project TimeGenerated, AccountType, Account, Computer = HostName, Type)
  );
  let pwrmd = PasswordResetMultiDataSource
  | project TimeGenerated, Computer, AccountType, Account, Type, TargetUserName;
  (union isfuzzy=true  
  (pwrmd
  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), Computerlist = make_set(Computer, 25), AccountType = make_set(AccountType, 25), Computer = arg_max(Computer , TimeGenerated), TargetUserList = make_set(TargetUserName, 25), TargetUserName = arg_max(TargetUserName, TimeGenerated), Total=count() by Account, Type
  | where Total > PerUserThreshold
  | extend ResetPivot = "PerUserReset"),  
  (pwrmd
  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ComputerList = make_set(Computer, 25), AccountList = make_set(Account, 25), AccountType = make_set(AccountType, 25), Account = arg_max(Account, TimeGenerated), Computer = arg_max(Computer , TimeGenerated), TargetUserList = make_set(TargetUserName,25), TargetUserName = arg_max(TargetUserName, TimeGenerated), Total=count() by Type
  | where Total > TotalThreshold
  | extend ResetPivot = "TotalUserReset")
  )
  | extend timestamp = StartTimeUtc, AccountCustomEntity = Account, HostCustomEntity = Computer
```

**Review by ashwin-patil — APPROVED** [review]

**Review by petebryan — APPROVED** [review]

**Pull request merged**